src/st/lib_psp/unk_1D9E8.c

```diff
@@ -7,7 +7,146 @@ INCLUDE_ASM("st/lib_psp/psp/lib_psp/unk_1D9E8", func_us_801BB8DC);
 
 INCLUDE_ASM("st/lib_psp/psp/lib_psp/unk_1D9E8", func_psp_09256018);
 
-INCLUDE_ASM("st/lib_psp/psp/lib_psp/unk_1D9E8", func_psp_092564B8);
+extern Collider D_us_80181AEC;
+extern u8 D_us_80181BCC;
+extern u8 D_us_80181BD8;
+extern u8 D_us_80181BE8;
+
+// EntityLesserDemonSpit code is shared with PSX and is ready for merge
+// Lesser Demon aerial spit projectile
+void EntityLesserDemonSpit(Entity* self) {
+    Primitive* prim;
+    s32 primIndex;
+    u16 posX, posY;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitLesserDemonSpit);
+        self->flags |= FLAG_DESTROY_IF_OUT_OF_CAMERA;
+        self->drawMode = DRAW_TPAGE2 | DRAW_TPAGE;
+        self->drawFlags |= FLAG_DRAW_ROTY | FLAG_DRAW_ROTX;
+        self->rotX = 0x180;
+        self->rotY = 0x180;
+        break;
+    case 1:
+        if (self->facingLeft) {
+            self->velocityX = FIX(4.0);
+        } else {
+            self->velocityX = FIX(-4.0);
+        }
+        self->velocityY = FIX(4.0);
+        self->step++;
+        break;
+    case 2:
+        MoveEntity();
+        AnimateEntity(&D_us_80181BCC, self);
+        if (CheckColliderOffsets(&D_us_80181AEC, 0)) {
+            self->animFrameIdx = 0;
+            self->animFrameDuration = 0;
+            self->rotX = 0x140;
+            self->rotY = 0x200;
+            self->hitboxWidth = 8;
+            self->hitboxHeight = 8;
+            self->step++;
+        }
+        break;
+    case 3:
+        if (!AnimateEntity(&D_us_80181BD8, self)) {
+            primIndex = g_api.AllocPrimitives(PRIM_GT4, 1);
+            if (primIndex != -1) {
+                self->flags |= FLAG_HAS_PRIMS;
+                self->primIndex = primIndex;
+                prim = &g_PrimBuf[primIndex];
+                self->ext.prim = prim;
```

Comment:
Not as clear here but probably this should be `self->ext.lesserDemon.unk7C` since you're using that extension in this file already